### PR TITLE
Refactor claims_match.

### PR DESCRIPTION
**Pull request by lwm**
- [x] Any changes relevant to users are recorded in the `CHANGELOG.md`.

---

~~@rohe, you'll have to advise on this one. I was just passing through
here and started to try to figure out what this function is really
doing.~~

OK, I think I've covered this refactoring.

### CODE REVIEW

File: src/oic/oic/__init__.py [hunk 1]
```diff
@@ -258,19 +258,17 @@ def add_token(self, resp):
...
-            if value not in val:
-                return False
-                # Whether it's essential or not doesn't change anything here
-    return True
+    :param claim1: First claim value
+    :param claim2: Second claim value
+    :return: Boolean
+    """
+    if isinstance(claim2, Iterable):
+        return claim1 in claim2
```
Comment (tpazderka) [inline src/oic/oic/__init__.py]: I think you need to do `claim1 in claim2.values()` to keep the logic of previous implementation.

Comment (lwm) [inline src/oic/oic/__init__.py]: Oh yes, better cover that too!

**Comment by tpazderka** [comment]
A test is missing for the `Iterable` claim and for simple `str` instances. Otherwise it looks good.

**Comment by lwm** [comment]
Missing tests added!

**Pull request merged**